R/addAlpha.R

```diff
@@ -0,0 +1,234 @@
+#' Estimate alpha diversity indices
+#' 
+#' The function estimates alpha diversity indices optionally using rarefaction,
+#' then stores results in \code{\link{colData}}.
+#' 
+#' @param x a \code{\link{SummarizedExperiment}} object.
+#' 
+#' @param assay.type the name of the assay used for calculation of the
+#'   sample-wise estimates (default: \code{assay.type = "counts"}).
+#'   
+#' @param index a \code{character} vector, specifying the alpha diversity 
+#'   indices to be calculated.
+#'   
+#' @param name a name for the column(s) of the colData the results should be
+#'   stored in. By default this will use the original names of the calculated
+#'   indices(By default: \code{name = index}).
+#'   
+#' @param ... optional arguments.
+#' 
+#' @param n.iter a single \code{integer} value for the number of rarefaction
+#'    rounds (By default: \code{n.iter = 10}).
```

Comment:
By default rarefaction should not be applied.

If n.iter=1 then rarefaction should be applied once at the given rarefaction depth. That would be the standard single-round rarification.

By default n.iter should be switched off. This could be n.iter=0, or n.iter=FALSE, or n.iter=NULL. Perhaps the last one n.iter=NULL would be the least ambiguous. Also n.iter=0 should lead to the same result, calculating alpha diversity without any rarefaction.

Add here the following explanation:  "Rarefaction is not applied by default (n.iter=NULL); if n.iter is a positive integer then rarefaction is applied (see @details section)." -> Then in @details explain how n.iter and rarefaction.depth operate.

Reply:
Unit tests should then check those conditions as well.